boto3/s3/transfer.py

```diff
@@ -184,15 +184,15 @@ def create_transfer_manager(client, config, osutil=None):
 
 def _should_use_crt(config):
     # This feature requires awscrt>=0.19.18
-    if HAS_CRT and has_minimum_crt_version((0, 19, 18)):
-        is_optimized_instance = awscrt.s3.is_optimized_for_system()
-    else:
-        is_optimized_instance = False
+    has_min_crt = HAS_CRT and has_minimum_crt_version((0, 19, 18))
+    is_optimized_instance = has_min_crt and awscrt.s3.is_optimized_for_system()
     pref_transfer_client = config.preferred_transfer_client.lower()
 
     if (
         is_optimized_instance
         and pref_transfer_client == constants.AUTO_RESOLVE_TRANSFER_CLIENT
+    ) or (
+        has_min_crt and pref_transfer_client == constants.CRT_TRANSFER_CLIENT
```

Comment:
How would you feel about adding some logic to `create_transfer_manager` that checks the CRT config and raises an exception if the CRT is set as the preferred transfer manager but it can't be loaded?  

That way a user won't have their config silently ignored if they forget to install the CRT on their system.

Reply:
Added check in https://github.com/boto/boto3/pull/4629/commits/a21de301a60572ab0e1c3f0e7dd30bf77400c3f5

Reply:
Oops, forgot to replace placeholder `Exception`s